File: Scroller1/src/main/java/com/example/scroller1/MainActivity.java
package com.example.scroller1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
   private TextView mText;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mText=findViewById(R.id.tv_context);
    }

    public void test(View view){
        View parent= (View) findViewById(R.id.tv_context).getParent();
//        parent.scrollBy(5,5);
        Log.e("TAG","scrolly=="+parent.getScrollY()+"===scrollx"+parent.getScrollX());
        parent.scrollTo(parent.getScrollX()+5,parent.getScrollY()+5);
    }
}

File: demo1/src/main/java/com/example/demo1/MainActivity.kt
package com.example.demo1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.SystemClock
import android.view.View
import android.widget.TextView
import com.example.demo.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val text=findViewById<TextView>(R.id.text)
        text.text= "taskid==$taskId"

    }

    fun click(view: View){
//        SystemClock.sleep(20*1000)
        val intent=Intent(this,MainActivity2::class.java)
        startActivity(intent)
    }
}